aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -376,8 +376,10 @@ TORCH_META_FUNC(amax)
 (const Tensor& self, IntArrayRef dim, bool keepdim) {
   auto maybe_result = maybe_get_output();
   if (maybe_result.defined()) {
-    TORCH_CHECK(self.scalar_type() == maybe_result.scalar_type(), "Expected the dtype for input and out to match, but got ",
-            self.scalar_type(), " for input's dtype and ",  maybe_result.scalar_type(), " for out's dtype.");
+    TORCH_CHECK(
+      self.scalar_type() == maybe_result.scalar_type(),
+      "Attempting to cast from ", self.scalar_type(), " to out tensor with dtype ", maybe_result.scalar_type(),
+      ", but this can't be cast because it is not safe!");
```

Comment:
I've changed this to use the error message from primTorch (instead of the other way around) because I think the primTorch one is more descriptive.

Reply:
While the primTorch dtype may be more descriptive, it doesn't seem correct in this situation. The check here requires the dtypes match, but the error message suggests the cast is checking for safe casting, instead.

Reply:
Resolved by making `amin` and `amax` refs to use `exact_dtype` and changing the error message a bit.